addons/web/static/src/js/core/sorted_registry.js

```diff
@@ -0,0 +1,64 @@
+odoo.define('web.SortedRegistry', function (require) {
+"use strict";
+
+var Registry = require('web.Registry');
+
+/**
+ * A sorted registry is a registry in whick items are sorted.
+ */
+var SortedRegistry = Registry.extend({
+    /**
+     * @constructor
+     * @param {Object} [mapping] the initial data in the registry
+     */
+    init: function () {
+        var self = this;
+        this._super.apply(this, arguments);
+        this._scores = [];
+        this._initialKeys = Object.keys(this.map);
+        this._sortedKeys = this._initialKeys;
+        this.listeners.push(function () {
+            self.sortedKeys = self._initialKeys.concat(
+                _.sortBy(self._scores, 'score').map(function (score) {
+                    return score.key;
+                })
+            );
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * overide
```

Comment:
not valid docstring. you should use `@override`. Also notice that it is not highlighted. Maybe your text editor is not properly configured